src/EditorFeatures/CSharpTest2/Recommendations/ReadOnlyKeywordRecommenderTests.cs

```diff
@@ -575,7 +570,18 @@ public async Task TestRefReadonlyInLocalFunction(bool topLevelStatement)
         public async Task TestRefReadonlyNotInRefExpression(bool topLevelStatement)
         {
             await VerifyAbsenceAsync(AddInsideMethod(
-@"ref int x = ref $$", topLevelStatement: topLevelStatement), options: CSharp9ParseOptions);
+@"ref int x = ref $$", topLevelStatement: topLevelStatement), options: CSharp9ParseOptions,
+                // Top level statement with script tested below, so skip it here.
+                scriptOptions: topLevelStatement ? CSharp9ParseOptions : null);
+        }
+
+        [CompilerTrait(CompilerFeature.ReadOnlyReferences)]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestRefReadonlyInRefExpression_TopLevelStatementScript()
+        {
+            // Recognized as parameter context, so readonly keyword is suggested.
```

Comment:
> Recognized as parameter context [](http://example.com/codeflow?start=14&length=32)

Is this expected and why?

Reply:
I didn't expect it, but it happens because `ref int x = ref ` is parsed as a method parameter in script mode with top level statements (not sure why, but that's unrelated to this change I think).